pkg/services/ngalert/notifier/alertmanager_test.go

```diff
@@ -346,3 +348,80 @@ func TestPutAlert(t *testing.T) {
 		})
 	}
 }
+
+// Tests cleanup of expired Silences. We rely on prometheus/alertmanager for
+// our alert silencing functionality, so we rely on its tests. However, we
+// implement a custom maintenance function for silences, because we snapshot
+// our data differently, so we test that functionality.
+func TestSilenceCleanup(t *testing.T) {
+	require := require.New(t)
+	assert := assert.New(t)
+
+	// TODO: We have to use a real clock because Prometheus doesn't support clock
+	// injection. So modify the retention constant. This isn't great - we should
+	// make this an argument to the alertmanager so we can safely change it.
```

Comment:
Unsure if this TODO is meant to be for this PR or the next one, but I would advise against leaving the TODO in the code if you don't plan to address this is your immediate next PR.

You already dropped a great issue on the Alertmanager repo, perhaps create one that references that in the Grafana side so that we can follow up?

Reply:
https://github.com/grafana/grafana/discussions/46822